### ISSUE TITLE
Disk Mounter applet has huge icons

### ISSUE THREAD

**Issue by N0rbert** [mate-desktop/mate-applets#457]
#### Expected behaviour
* Disk Mounter applet has icons with normal sizes

#### Actual behaviour
* Disk Mounter icons are huge - do not fit to default 28 px height (see screenshot)

![USB is oversized](https://launchpadlibrarian.net/480709587/disk-mounter-2004.png)

#### Steps to reproduce the behaviour
1. Have Ubuntu MATE 20.04 LTS installed and logged in MATE session
2. Add Disk Mounter applet to the MATE Panel

#### MATE general version
1.24

#### Package version
1.24.0-1

#### Linux Distribution
Ubuntu MATE 20.04 LTS

#### Link to downstream report of your Distribution
https://bugs.launchpad.net/bugs/1863436

**Comment by raveit65** [comment]
No problem with fedora 32 , Mate 1.24 and mate-icon-theme!
![Bildschirmfoto zu 2020-05-21 10-46-18](https://user-images.githubusercontent.com/961604/82541271-7a506700-9b50-11ea-8c14-0e4c682ea0c4.png)
Have you ever tried to us Mate default icon-theme which is mate-icon-theme?

Btw. Default pannel size is 
```
[rave@mother ~]$ gsettings reset org.mate.panel.toplevel:/ size
[rave@mother ~]$ gsettings get org.mate.panel.toplevel:/ size
24
```

**Comment by N0rbert** [comment]
Thanks, but I'm not really alone, other user [has this problem](https://ubuntu-mate.community/t/drive-icons-large-distorted-in-20-04/21874?u=norbert_x) too.
Really - the 24 or 28 px does not matter, as USB drive icon does not fit using default settings:

![Texaholic's screenshot](https://i.imgur.com/KMOwh3q.jpg)

Maybe it is [SVG scaling problem](https://ubuntu-mate.community/t/panel-bugs-and-clementine-icon-is-too-big/15880/20?u=norbert_x), I'm not sure.

**Comment by raveit65** [comment]
I am only wondering that i can't reproduce it with fedora.
So, we need to find out what is different.
Can you please try mate-icon-theme which use PNGs?
And the applet is on the bottom panel?

**Comment by N0rbert** [comment]
Thanks, setting PNG theme helps:

`gsettings set org.mate.interface icon-theme 'mate'`

results in:

![mate-theme](https://user-images.githubusercontent.com/309456/82579394-692a4900-9b96-11ea-9b42-6a014cb68dda.png)

But other themes has issues:

| Theme                         |  Look  |
|:------------------------------|:-------|
| Ambiant-MATE                  | fail   |
| Breathe                       | ok     |
| breeze                        | ok     |
| breeze-dark                   | ok     |
| ContrastHigh                  | ok     |
| elementary                    | fail   |
| ElementaryEdubuntu            | fail   |
| elementary-mono-dark          | fail   |
| elementary-xfce               | fail   |
| elementary-xfce-dark          | fail   |
| elementary-xfce-darker        | fail   |
| elementary-xfce-darkest       | fail   |
| elementaryXubuntu-dark        | fail   |
| Enlightenment-X               | ok     |
| ePapirus                      | ok     |
| GartoonRedux                  | ok     |
| gnome                         | ok     |
| hicolor                       | ok     |
| Human                         | ok     |
| Humanity                      | fail   |
| Humanity-Dark                 | fail   |
| HumanLoginIcons               | fail   |
| LoginIcons                    | fail   |
| mate                          | ok     |
| menta                         | ok     |
| nuoveXT2                      | ok     |
| oxygen                        | ok     |
| Papirus                       | ok     |
| Papirus-Dark                  | ok     |
| Papirus-Light                 | ok     |
| Radiant-MATE                  | fail   |
| Tango                         | ok     |
| ubuntu-mono-dark              | fail   |
| ubuntu-mono-light             | fail   |
| ubuntustudio                  | ok     |
| ubuntustudio-dark             | ok     |
| unity-icon-theme              | fail   |
| Yaru                          | ok     |

**Comment by raveit65** [comment]
Ok, calling a Mate developer which use Ubuntu. Probably he has an Idea.
@flexiondotorg

**Comment by raveit65** [comment]
There was an change in 2018 with the icons, But this is already in mate-1.22. So, i am not sure if this is related?
https://github.com/mate-desktop/mate-applets/pull/318
@vkareh  ^^^^^

@N0rbert 
Does the problem exists with previous Ubuntu version and Mate 1.22?

**Comment by N0rbert** [comment]
Here are test results:
| Ubuntu version | mate-applets version | Look with default Ambiant-MATE theme |
|:--------------------|:---------------------------|:---------------------------------------------------|
| 18.04 LTS | 1.20.1-3 | ok |
| 18.10 | 1.20.2-2 | fail |
| 19.04 | 1.20.3-2 | fail |
| 19.10 | 1.22.2-0ubuntu1 | fail |
| 20.04 LTS | 1.24.0-1 | fail |

**Comment by raveit65** [comment]
Ok, than it seems to be a side effect of https://github.com/mate-desktop/mate-applets/pull/318
@vkareh 
Please ^^^^

**Comment by marosg42** [comment]
I see this on 20.04 with Ambient-MATE as well, I did not encounter it on 18.04. Switching icon-theme helps as mentioned above.
I did some code debugging. If the panel size is up to 24 pixels, all is fine, once it changes to 25, icon jumps to much bigger.

It looks like the problem happens when 
`gtk_icon_theme_lookup_by_gicon_for_scale()` is called, it uses correct parameters but returned icon is probably wrong size, I am not sure how to check.

Maybe issue in svg files? But then I don't understand why it does not happen in Fedora.

**Comment by marosg42** [comment]
So, could it be this? I did not find the correct icon in Ambiant-MATE icons, it looks like it loads `/usr/share/icons/Humanity/devices/24/drive-removable-media-usb.svg`. (I think the actual filename which is loaded could be figured out in the code, I will check later).

`drive-removable-media-usb.svg` is missing in `/usr/share/icons/Humanity/devices/32/` and then it exists again in `/usr/share/icons/Humanity/devices/48/`. And icon is ugly when panel is from 25 to around 46 pixels high.

**Comment by marosg42** [comment]
Yes, it is using Humanity icons. And it jumps from 24 to 48 which is a problem

**Comment by marosg42** [comment]
Tested the same with 18.04. It also does not have `drive-removable-media-usb.svg` in 32 directory, but scaling works just fine. It must be GTK library difference

**Comment by lukefromdc** [comment]
I have not seen this happen with any of the MATE icon themes. Note that some GTK 3 versions have had issues with large format icons (with smaller icons not included)  in any menu and in some other places and refused to scale them down.
